foursight_core/app_utils.py

```diff
@@ -334,6 +547,175 @@ def trim_output(cls, output, max_size=100000):
             return cls.TRIM_ERR_OUTPUT
         return output
 
+    def sorted_dict(self, dictionary: dict) -> dict:
+        result = {}
+        for key in sorted(dict(dictionary).keys(), key=lambda key: key.lower()):
+            result[key] = dictionary[key]
+        return result
+
+    def get_aws_account_number(self) -> dict:
+        try:
+            caller_identity = boto3.client("sts").get_caller_identity()
+            return caller_identity["Account"]
+        except Exception:
+            logger.warn("Cannot determine AWS credentials token (not fatal - just for Foursight UI display).")
+            return None
+
+    def get_obfuscated_credentials_info(self, env_name: str, stage_name: str) -> dict:
+        try:
+            session = boto3.session.Session()
+            credentials = session.get_credentials()
+            access_key_id = credentials.access_key
+            region_name = session.region_name
+            caller_identity = boto3.client("sts").get_caller_identity()
+            user_arn = caller_identity["Arn"]
+            account_number = caller_identity["Account"]
+            auth0_client_id = self.get_auth0_client_id(env_name, stage_name)
+            credentials_info = {
+                "AWS Account Number:": account_number,
+                "AWS User ARN:": user_arn,
+                "AWS Access Key ID:": access_key_id,
+                "AWS Region Name:": region_name,
+                "Auth0 Client ID:": auth0_client_id
+            }
+            return credentials_info
+        except Exception as e:
+            logger.warn("Cannot determine AWS credentials token (not fatal - just for Foursight UI display).")
+            logger.warn(e)
+            return {}
+
+    def convert_utc_datetime_to_useastern_datetime(self, t: str) -> str:
```

Comment:
I'm not sure this is correct. Strings should only be used for output, not as part of conversion. You want perhaps to take advantage of operations like these. You'll have to untangle them into your specific need:
```
>>> import pytz, datetime
>>> str(datetime.datetime.utcnow().replace(tzinfo=pytz.UTC).astimezone(pytz.timezone("US/Eastern")))
'2022-08-22 12:18:41.840687-04:00'
>>> str(datetime.datetime.now())
'2022-08-22 12:18:48.728836'
```
Note that "US/Eastern" is a special thing that knows about daylight savings shifts. You probably already realize this but if the difference between -0400 and -0500 is not clear to you, ask me.
It is fair to want to strip the -nnnn that results and put a specific string on the end. I'm not sure why America/New_York is getting involved here, but I think that is not needed. US/Eastern is more conventional. Only in places like Arizona that are jurisdiction-specific do you need to refer to a city or state.

Reply:
The reason I'm parsing a date string *input* is because it's used for getting the lambda modified time, which boto3 returns as a datetime *string*; this is just for informational purposes - can be useful to know when the lambda was last deployed). Thanks for the better way of doing the conversion.